### PR TITLE
Export Rule type to consumers

### PR DESCRIPTION
**Pull request by MarceloPrado**
Add missing `Rule` export. I also added some docs to improve DX when importing it:
CleanShot 2023-07-01 at 12 37 25@2x

### CODE REVIEW

**Review by decs — CHANGES_REQUESTED** [review]
Nice, this is looking good! Could you also run `yarn lint`? I'll look into how to make these scripts run automatically.

File: src/rules/index.ts [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
 };
+
+/**
+ * Allows you to define complex conditions and criteria for decision-making. It
+ * consists of one or more signals, which can be combined using logical
+ * operators to create intricate structures.
+ * 
+ * Takes a TContext argument which encapsulates the necessary information
+ * required by signals to make decisions and determine the outcome of rules.
+ */
```
Comment (decs) [inline src/rules/index.ts:28]: What do you think of moving this comment to the `Rule` class? That way we wouldn't need to define the internal `RuleType`. And the type can be re-exported with `export type {default as Rule} from './rule';`

Comment (MarceloPrado) [reply src/rules/index.ts]: Great idea! Addressed in [1d6ecac](https://github.com/decs/ruls/pull/2/commits/1d6ecac4b6ac81e5628c6ea1bc4cec1f3c672601) 🙂

**Comment by MarceloPrado** [comment]
@decs cool, that would be great! We could have a basic GH action that runs `format` + `lint`, and one for auto publishing changes
Addressed your feedbacks !

**Review by decs — APPROVED** [review]
Let's go!

**Comment by decs** [comment]
Published as `v1.0.2`

**Comment by decs** [comment]
> @decs cool, that would be great! We could have a basic GH action that runs `format` + `lint`, and one for auto publishing changes Addressed your feedbacks !

I set up this automation! Now, the code is automatically linted, formatted, and tested before committing. And landing a commit with an updated version on `package.json` will auto-publish it on npm 🤩

**Comment by MarceloPrado** [comment]
Incredible! Great work

**Pull request merged**